### PR TITLE
Add readLinesAll and readUtf8All to LocationIO

### PR DESCRIPTION
**Pull request by raronson**
@markhibberd @nhibberd I found i have been doing this a lot in tests so thought i should move it into the lib

### CODE REVIEW

File: notion-core/src/test/scala/com/ambiata/notion/core/LocationIOSpec.scala [hunk 1]
```diff
@@ -123,7 +142,26 @@ class LocationIOSpec extends Specification with ScalaCheck { def is = s2"""
...
+      }
+
+    From a dir/prefix using readUtf8All
+
+     ${ prop((loc: LocationTemporary, str1: S, str2: S) => (for {
+          p <- LocationIO.fromRIO(loc.location)
+          _ <- LocationIO.writeUtf8(p | Component("1"), str1.value)
+          _ <- LocationIO.writeUtf8(p | Component("2"), str2.value)
+          r <- LocationIO.readUtf8All(p)
+        } yield r) must beOkValue(loc.context)(str1.value + str2.value))
```
Comment (nhibberd) [inline notion-core/src/test/scala/com/ambiata/notion/core/LocationIOSpec.scala]: Should be sorting both sides here?

File: notion-core/src/test/scala/com/ambiata/notion/core/LocationIOSpec.scala [hunk 2]
```diff
@@ -86,7 +86,26 @@ class LocationIOSpec extends Specification with ScalaCheck { def is = s2"""
...
+      }
+
+    From a dir/prefix using readLinesAll
+
+     ${ prop((loc: LocationTemporary, l1: List[N], l2: List[N]) => (for {
+          p <- LocationIO.fromRIO(loc.location)
+          _ <- LocationIO.writeUtf8Lines(p | Component("1"), l1.map(_.value))
+          _ <- LocationIO.writeUtf8Lines(p | Component("2"), l2.map(_.value))
+          r <- LocationIO.readLinesAll(p)
+        } yield r) must beOkValue(loc.context)((l1 ++ l2).map(_.value)))
```
Comment (nhibberd) [inline notion-core/src/test/scala/com/ambiata/notion/core/LocationIOSpec.scala]: Sort here as well

**Comment by nhibberd** [comment]
:+1: on :green_apple:

File: notion-core/src/test/scala/com/ambiata/notion/core/LocationIOSpec.scala [hunk 3]
```diff
@@ -77,7 +77,7 @@ class LocationIOSpec extends Specification with ScalaCheck { def is = s2"""
 
   LocationIO should be able to read/write lines from a location
 
-    From a sinlge file/object
+    From a sinlge file/object using readLines
```
Comment (HuwCampbell) [inline notion-core/src/test/scala/com/ambiata/notion/core/LocationIOSpec.scala]: "single"

**Comment by raronson** [comment]
Closing, this isn't as useful or generic as i thought

**Pull request closed without merging**